### DISCUSSION TITLE
BingX-Python

### DISCUSSION THREAD

**Discussion by HisokaGit** [community/community#73610] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Alguien sabe cómo se puede fijar el apalancamiento en en futuros perpetuos usando la API de BingX?

**Comment by TheEdaL06** [comment, marked as answer]
Los futuros perpetuos de BingX utilizan un sistema de "Apalancamiento escalonado". [Con este sistema, los límites de posición se ajustarán según el apalancamiento seleccionado por el usuario](https://support.bingx.com/hc/es/articles/360045899754-Futuros-Perpetuo-Pares-de-trading-apalancamiento-y-l%C3%ADmites). Los usuarios pueden obtener un tamaño de posición mayor al seleccionar un apalancamiento inferior.

Para ajustar el apalancamiento a través de la API de BingX, necesitarás consultar la documentación oficial de la API de BingX. En la documentación, deberías encontrar detalles sobre cómo interactuar con la API, incluyendo cómo establecer el apalancamiento para los futuros perpetuos.

Si no puedes encontrar la información específica que necesitas en la documentación, te recomendaría que te pongas en contacto con el servicio de soporte de BingX. Ellos deberían ser capaces de proporcionarte más detalles o guiarte a través del proceso. Recuerda que el apalancamiento y los límites de posición están influenciados por el nivel de riesgo del usuario, que puede variar entre usuarios.

**Reply by HisokaGit** [reply]
En la documentación de la API no hay información al respecto, estoy tratando de ponerme en contacto con soporte

**Comment by MatiasGabrielPaz** [comment]
Hola, este es el ejemplo que da BingX para cambiar el apalancamiento:

import time
import requests
import hmac
from hashlib import sha256

APIURL = "https://open-api.bingx.com";
APIKEY = ""
SECRETKEY = ""

def demo():
    payload = {}
    path = '/openApi/swap/v2/trade/leverage'
    method = "GET"
    paramsMap = {
    "symbol": "BTC-USDT",
    "recvWindow": 0
}
    paramsStr = praseParam(paramsMap)
    return send_request(method, path, paramsStr, payload)

def get_sign(api_secret, payload):
    signature = hmac.new(api_secret.encode("utf-8"), payload.encode("utf-8"), digestmod=sha256).hexdigest()
    print("sign=" + signature)
    return signature


def send_request(method, path, urlpa, payload):
    url = "%s%s?%s&signature=%s" % (APIURL, path, urlpa, get_sign(SECRETKEY, urlpa))
    print(url)
    headers = {
        'X-BX-APIKEY': APIKEY,
    }
    response = requests.request(method, url, headers=headers, data=payload)
    return response.text

-----------

Fijate en la documentacion > Trade Interface > Query Leverage.

Te paso mi Discord user: bunnydroid. Yo tambien estoy haciendo un proyecto con la api y estoy teniendo algunos issues. Capaz nos podemos ayudar.

**Comment by ahaha2023** [comment]
Hola, este es el ejemplo que da BingX para cambiar el apalancamiento:

import time
import requests
import hmac
from hashlib import sha256

APIURL = "https://open-api.bingx.com/";
APIKEY = ""
SECRETKEY = ""

def demo():
payload = {}
path = '/openApi/swap/v2/trade/leverage'
method = "GET"
paramsMap = {
"symbol": "BTC-USDT",
"recvWindow": 0
}
paramsStr = praseParam(paramsMap)
return send_request(method, path, paramsStr, payload)

def get_sign(api_secret, payload):
signature = hmac.new(api_secret.encode("utf-8"), payload.encode("utf-8"), digestmod=sha256).hexdigest()
print("sign=" + signature)
return signature

def send_request(method, path, urlpa, payload):
url = "%s%s?%s&signature=%s" % (APIURL, path, urlpa, get_sign(SECRETKEY, urlpa))
print(url)
headers = {
'X-BX-APIKEY': APIKEY,
}
response = requests.request(method, url, headers=headers, data=payload)
return response.text

Fijate en la documentacion > Trade Interface > Query Leverage.

Te paso mi Discord user: bunnydroid. Yo tambien estoy haciendo un proyecto con la api y estoy teniendo algunos issues. Capaz nos podemos ayudar.
